R/stan_betareg.R

```diff
@@ -121,7 +121,12 @@ stan_betareg <- function (formula, data, subset, na.action, weights, offset,
   offset <- validate_offset(as.vector(model.offset(mf)), y = Y)
   if (!length(prior_ops)) 
     prior_ops <- list(scaled = FALSE, prior_scale_for_dispersion = Inf)
-
+  
+  # determine whether user specified regression matrix for precision model
+  if (length(grep("\\|", all.names(formula))) == 1 && length(link.phi) > 1) {
```

Comment:
Ok but now I'm confused why this is here but `match.arg(link)` and `match.arg(link.phi)` occur below _after_ fitting the model.  I think `link` and `link.phi` should be validated before passing them on to another function (`stan_betareg.fit`). And simply doing `link.phi <- match.arg(link.phi)` will take care of ensuring that `link.phi` has length 1 (it will be `"log"` if user leaves the default).